src/main/scala/com/metl/model/BackendAdaptor.scala

```diff
@@ -550,3 +551,28 @@ class ResourceCachingAdaptor(sc:ServerConfiguration,cacheConfig:CacheConfig) ext
     super.isReady
   }
 }
+
+class D2LCachingAdaptor(cacheConfig:CacheConfig) {
```

Comment:
hmm... with regards the "premature optimization" comment:

a) it's in prod
b) when you describe "realtime view over changing data", do you envisage it being queried regularly to watch the data change in realtime?
c) right now it may be realtime data, but it could be temporally inconsistent if half the data is cached and the other half is current.

We might want to intentionally cache the entire output so that it's temporally consistent, and then simply provide a short expiry time, so that we're protecting ourselves from re-entrancy.  How about a 1 minute expiry and cache the whole thing?